indra/newview/llvoavatar.cpp

```diff
@@ -8097,21 +8106,24 @@ bool LLVOAvatar::getIsCloud() const
 			);
 }
 
-void LLVOAvatar::updateRezzedStatusTimers(S32 rez_status)
+void LLVOAvatar::updateRezzedStatusTimers(ERezzedStatus rez_status)
 {
 	// State machine for rezzed status. Statuses are -1 on startup, 0
 	// = cloud, 1 = gray, 2 = downloading, 3 = waiting for attachments, 4 = full.
 	// Purpose is to collect time data for each it takes avatar to reach
 	// various loading landmarks: gray, textured (partial), textured fully.
 
+    auto next = [](ERezzedStatus status) -> ERezzedStatus { return (ERezzedStatus)(status + 1); };
+    auto incr = [&](ERezzedStatus& status) { status = next(status); };
```

Comment:
This looks needlesly overcomplicated. I suggest just leaving it as a loop over S32